src/librustdoc/html/static/css/rustdoc.css

```
@@ -735,17 +735,21 @@ a {
 .anchor {
 	display: none;
 	position: absolute;
-	left: -7px;
+	left: 0;
+	background: none !important;
 }
 .anchor.field {
 	left: -5px;
 }
 .small-section-header > .anchor {
-	left: -28px;
-	padding-right: 10px; /* avoid gap that causes hover to disappear */
+	left: -15px;
+	padding-right: 8px;
 }
-.anchor:before {
-	content: '\2002\00a7\2002';
+h2.small-section-header > .anchor {
+	padding-right: 2px;
```

It looks like maybe it could be fixed by increasing this padding?

---

It seems like just increasing this to 6px fixes the problem.

```suggestion
	padding-right: 6px;
```